src/gstools/field/generator.py

```diff
@@ -532,3 +538,332 @@ def _create_unit_vector(self, broadcast_shape, axis=0):
+        self.update(model, seed, mode_no, period)
+
+    def __call__(self, pos, add_nugget=True):
+        """Calculate the modes for the Fourier method.
+
+        This method  calls the `summate_fourier` Cython method, which is the
+        heart of the Fourier method.
+
+        Parameters
+        ----------
+        pos : (d, n), :class:`numpy.ndarray`
+            the position tuple with d dimensions and n points.
+        add_nugget : :class:`bool`
+            Whether to add nugget noise to the field.
+
+        Returns
+        -------
+        :class:`numpy.ndarray`
+            the random modes
+        """
+        pos = np.asarray(pos, dtype=np.double)
+
+        summed_modes = summate_fourier(
+            self._spectrum_factor,
+            self._modes,
+            self._z_1,
+            self._z_2,
+            pos,
+            config.NUM_THREADS,
+        )
+        nugget = self.get_nugget(summed_modes.shape) if add_nugget else 0.0
+        return summed_modes + nugget
+
+    def get_nugget(self, shape):
+        """
+        Generate normal distributed values for the nugget simulation.
+
+        Parameters
+        ----------
+        shape : :class:`tuple`
+            the shape of the summed modes
+
+        Returns
+        -------
+        nugget : :class:`numpy.ndarray`
+            the nugget in the same shape as the summed modes
+        """
+        if self.model.nugget > 0:
+            nugget = np.sqrt(self.model.nugget) * self._rng.random.normal(
+                size=shape
+            )
+        else:
+            nugget = 0.0
+        return nugget
+
+    def update(self, model=None, seed=np.nan, mode_no=None, period=None):
+        """Update the model and the seed.
+
+        If model and seed are not different, nothing will be done.
+
+        Parameters
+        ----------
+        model : :any:`CovModel` or :any:`None`, optional
+            covariance model. Default: :any:`None`
+        seed : :class:`int` or :any:`None` or :any:`numpy.nan`, optional
+            the seed of the random number generator.
+            If :any:`None`, a random seed is used. If :any:`numpy.nan`,
+            the actual seed will be kept. Default: :any:`numpy.nan`
+        mode_no : :class:`list` or :any:`None`, optional
+            Number of Fourier modes per dimension.
+        period : :class:`list` or :any:`None, optional
+            The spatial periodicity of the field, is often the domain size.
+        """
+        tmp_model = model if model is not None else self._model
+        if period is not None:
+            if len(period) != tmp_model.dim:
+                raise ValueError(
+                    "Fourier: Dimension mismatch in argument period."
+                )
+            self._period = np.array(period)
+            self._delta_k = 2.0 * np.pi / self._period
+            if mode_no is None:
+                self._set_modes(self._mode_no, tmp_model)
+        if mode_no is not None:
```

Comment:
There should be an else branch checking if `self._mode_no` is None to ensure mode_no is given.

Reply:
I'm not sure, if I understand you correctly, but it should be a valid way to call the `update` , e.g. if you only want to change the `period`, but not the `mode_no`. That case is already handled by the code.

Reply:
Now that we have the `Fourier.__init__` argument `mode_no` set to a default value, this shouldn't be possible anymore, right? - I mean, except someone tries really hard ;-)

> or on the verge to start using static typing in python ;-)

No way! :rofl: You just proposed to make it possible to do `period=200` *and* `period=[200, 200]`.

Reply:
OK, you are right. Btw, that is totally fine with typing and would be `Union[int, List[int]]` ;-)